context/python/parse.py

```diff
@@ -7,62 +7,73 @@
 from sets import Sets
 
 
+def arg_parser():
+    parser = argparse.ArgumentParser(
+        description='A variety of set intersection visualizations')
+
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument(
+        '--json', type=str,
+        help='If given file does not exist, fall back to environment variable')
+    group.add_argument(
+        '--lists', nargs='+', type=argparse.FileType('r'),
+        help='Single column lists')
+
+    parser.add_argument(
+        '--output', type=str,
+        help='Destination directory')
+
+    return parser
+
+
 def get_input_json(possible_input_file):
     '''
     Checks three possible sources for JSON input,
     and returns JSON string if found.
     Returns None if all three fail.
     '''
-    if possible_input_file and os.path.isfile(possible_input_file):
+    if os.path.isfile(possible_input_file):
         return open(possible_input_file, 'r').read(None)
 
     json = os.environ.get('INPUT_JSON')
     if json:
+        print('INPUT_JSON: ' + json[:40])
         return json
 
     url = os.environ.get('INPUT_JSON_URL')
     if url:
-        print('url: ' + url)
+        print('INPUT_JSON_URL: ' + url)
         return requests.get(url).text
```

Comment:
I forget if `.text` also yields proper JSON or if you have to do a `.json()` here

Reply:
This method just returns a string, and json.loads is downstream. Don't feel super strongly about it, but this way deserialization is done in only one place... One black box instead of two.